lua/starfall/transfer.lua

```diff
@@ -106,6 +107,34 @@ if SERVER then
 		return true
 	end
 
+	function SF.SendError(chip, message, traceback, client, should_notify)
+		net.Start("starfall_error")
+			net.WriteEntity(chip)
+			net.WriteString(string.sub(chip.sfdata.mainfile, 1, 1024)
+			net.WriteString(string.sub(message, 1, 1024)
+			net.WriteString(string.sub(traceback, 1, 1024)
+		if client~=nil and should_notify~=nil then
+			net.WriteBool(true)
+			net.WriteEntity(client)
+			net.WriteBool(should_notify)
+			net.SendOmit(client)
+		else
+			net.WriteBool(false)
+			net.Broadcast()
+		end
+	end
+
+	net.Receive("starfall_error", function(_, ply)
+		local chip = net.ReadEntity()
+		if not (chip and chip:IsValid()) then return end
```

Comment:
Same as before, check whether the `chip.owner` is valid or allow `StarfallError` to run with an invalid `owner`.